src/core/vdom/helpers/normalize-children.js

```diff
@@ -45,7 +45,9 @@ function normalizeArrayChildren (children: any, nestedIndex?: string): Array<VNo
     last = res[res.length - 1]
     //  nested
     if (Array.isArray(c)) {
-      res.push.apply(res, normalizeArrayChildren(c, `${nestedIndex || ''}_${i}`))
+      // avoid elements to be set key, which created by slot
+      const subNestedIndex = isTrue(c._rendered) ? undefined : `${nestedIndex || ''}_${i}`
```

Comment:
`_renndered` is set when `process.env.NODE_ENV !== 'production'`.

and not only slot, hand-written render functions could have this problem too.